core/src/main/java/org/apache/lucene/queryparser/classic/QueryParserSettings.java

```diff
@@ -19,64 +19,72 @@
 
 package org.apache.lucene.queryparser.classic;
 
-import com.carrotsearch.hppc.ObjectFloatHashMap;
 import org.apache.lucene.analysis.Analyzer;
-import org.apache.lucene.search.FuzzyQuery;
 import org.apache.lucene.search.MultiTermQuery;
-import org.apache.lucene.util.automaton.Operations;
 import org.elasticsearch.common.unit.Fuzziness;
 import org.joda.time.DateTimeZone;
 
-import java.util.List;
 import java.util.Locale;
+import java.util.Map;
 
 /**
- *
+ * Encapsulates settings that affect query_string parsing via {@link MapperQueryParser}
  */
 public class QueryParserSettings {
```

Comment:
just a crazy idea mabye we can have the query builder subclass this class and have all the settings in one place?